build.gradle

```diff
@@ -3,7 +3,7 @@ plugins {
     id 'maven'
     id 'idea'
     id 'application'
-    id 'org.springframework.boot' version '2.7.12'
+    id 'org.springframework.boot' version '3.1.2'
```

Comment:
I'm not sure of the limitations between the versions, but is there reason why the dependency is using 3.1.2 instead of the most recent 3.1.4 (9/21/23)?

Reply:
Thank you for your review. This PR works is in progress. 
Also yes, as you pointed, we are updating to 3.1.4 spring along with 7.6 gradle wrapper version.

Reply:
Updated latest changes.